### PR TITLE
Fix WALInputStream read ByteBuffer issues

### PR DESCRIPTION
**Pull request by shuwenwei**
## Description
Fix WALInputStream read ByteBuffer issues

### CODE REVIEW

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALInputStream.java [hunk 1]
```diff
@@ -312,12 +312,16 @@ public void skipToGivenLogicalPosition(long pos) throws IOException {
 
   public void read(ByteBuffer buffer) throws IOException {
     int totalBytesToBeRead = buffer.remaining();
-    int currReadBytes = Math.min(dataBuffer.remaining(), buffer.remaining());
-    dataBuffer.get(buffer.array(), buffer.position(), currReadBytes);
-    if (totalBytesToBeRead - currReadBytes > 0) {
-      loadNextSegment();
-      read(buffer);
+    while (totalBytesToBeRead > 0) {
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALInputStream.java:315]: add some test?

Comment (shuwenwei) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALInputStream.java:315]: Added

**Review by jt2594838 — APPROVED** [review]

File: iotdb-core/datanode/src/test/java/org/apache/iotdb/db/storageengine/dataregion/wal/compression/WALCompressionTest.java [hunk 2]
```diff
@@ -162,6 +162,38 @@ public void testSkipToGivenPosition()
...
+      positionAndEntryPairList.add(new Pair<>(fileOffset, buffer.position()));
+      fileOffset += buffer.position();
+      writer.write(buffer);
+      buffer.clear();
+    }
+    writer.close();
+
+    try (WALInputStream stream = new WALInputStream(walFile)) {
+      stream.skipToGivenLogicalPosition(positionAndEntryPairList.get(0).left);
+      ByteBuffer buffer1 =
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/test/java/org/apache/iotdb/db/storageengine/dataregion/wal/compression/WALCompressionTest.java:185]: Smark test！

Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/test/java/org/apache/iotdb/db/storageengine/dataregion/wal/compression/WALCompressionTest.java:185]: Smart Test!

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**